search/ranker.cpp

```diff
@@ -337,12 +349,19 @@ class RankerResultMaker
         {
           auto const type = Model::TYPE_CITY;
           auto const & range = preInfo.m_tokenRange[type];
-          errorsMade += GetErrorsMade(*city, m_params, range, type);
+          auto const matchingResult = MatchTokenRange(*city, m_params, range, type);
+          errorsMade += matchingResult.first;
+          matchedLength += matchingResult.second;
         }
       }
 
+      size_t totalLegth = 0;
+      for (size_t i = 0; i < m_params.GetNumTokens(); ++i)
+        totalLegth += m_params.GetToken(i).GetOriginal().size();
+
       info.m_nameScore = nameScore;
       info.m_errorsMade = errorsMade;
+      info.m_matchedPart = static_cast<double>(matchedLength) / static_cast<double>(totalLegth);
```

Comment:
please set to zero if totalLength is zero

Reply:
done